services/blockstorage/service.go

```diff
@@ -120,13 +208,37 @@ func (s *service) validateBlockDoesNotExist(txBlockHeader *protocol.Transactions
 	return true
 }
 
-func (s *service) validateProtocolVersion(txBlockHeader *protocol.TransactionsBlockHeader) error {
+func (s *service) validateBlockHeight(blockPair *protocol.BlockPairContainer) error {
+	expectedBlockHeight := s.lastCommittedBlockHeight + 1
+
+	txBlockHeader := blockPair.TransactionsBlock.Header
+	rsBlockHeader := blockPair.ResultsBlock.Header
+
+	if txBlockHeader.BlockHeight() != expectedBlockHeight {
+		return fmt.Errorf("block height is %d, expected %d", txBlockHeader.BlockHeight(), expectedBlockHeight)
+	}
+
+	if rsBlockHeader.BlockHeight() != expectedBlockHeight {
+		return fmt.Errorf("block height is %d, expected %d", rsBlockHeader.BlockHeight(), expectedBlockHeight)
+	}
+
+	return nil
+}
+
+func (s *service) validateProtocolVersion(blockPair *protocol.BlockPairContainer) error {
+	txBlockHeader := blockPair.TransactionsBlock.Header
+	rsBlockHeader := blockPair.ResultsBlock.Header
+
 	if txBlockHeader.ProtocolVersion() != ProtocolVersion {
 		err := fmt.Errorf("protocol version mismatch: expected 1 got %d", txBlockHeader.ProtocolVersion())
```

Comment:
change expected 1 to expected %d and use the constant (this is actually my mistake from last week i think)

Reply:
All of it will change when we merge #33